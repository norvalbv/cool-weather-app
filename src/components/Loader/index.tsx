import React, { ReactElement } from 'react';
import LoaderIcon from 'components/SVG/LoaderIcon';

const Loader = (): ReactElement => {
  return (
    <div className="grid h-screen w-full place-items-center">
      <LoaderIcon />
    </div>
  );
};

export default Loader;
