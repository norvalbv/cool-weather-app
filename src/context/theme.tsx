import { ReactElement, createContext, useState } from 'react';

type ThemeContextValue = {
  isDarkMode: boolean;
  toggleDarkMode: () => void;
};

const ThemeContext = createContext<ThemeContextValue | null>(null);

type ThemeProviderProps = {
  children: JSX.Element | JSX.Element[];
};

export const ThemeContextProvider = ({ children }: ThemeProviderProps): ReactElement => {
  const [isDarkMode, setIsDarkMode] = useState(false);

  const toggleDarkMode = (): void => setIsDarkMode(!isDarkMode);

  return (
    <ThemeContext.Provider value={{ isDarkMode, toggleDarkMode }}>{children}</ThemeContext.Provider>
  );
};

export default ThemeContext;
