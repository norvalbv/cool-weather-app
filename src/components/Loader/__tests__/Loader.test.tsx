import React from 'react';
import renderer from 'react-test-renderer';

import { ThemeContextProvider } from 'context/theme';

import Loader from '..';

describe('<Loader />', () => {
  test('Should render correctly with mandatory props', () => {
    const tree = renderer
      .create(
        <ThemeContextProvider>
          <Loader />
        </ThemeContextProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
