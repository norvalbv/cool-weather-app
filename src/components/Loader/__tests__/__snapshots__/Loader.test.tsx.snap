// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader /> Should render correctly with mandatory props 1`] = `
<div
  className="grid h-screen w-full place-items-center"
>
  <svg
    height="57"
    stroke="#d4d4d4"
    viewBox="0 0 57 57"
    width="57"
    xmlns="http://www.w3.org/2000/svg"
  >
    <g
      fill="none"
      fillRule="evenodd"
    >
      <g
        strokeWidth="2"
        transform="translate(1 1)"
      >
        <circle
          cx="5"
          cy="50"
          r="5"
        >
          <animate
            attributeName="cy"
            begin="0s"
            calcMode="linear"
            dur="2.2s"
            repeatCount="indefinite"
            values="50;5;50;50"
          />
          <animate
            attributeName="cx"
            begin="0s"
            calcMode="linear"
            dur="2.2s"
            repeatCount="indefinite"
            values="5;27;49;5"
          />
        </circle>
        <circle
          cx="27"
          cy="5"
          r="5"
        >
          <animate
            attributeName="cy"
            begin="0s"
            calcMode="linear"
            dur="2.2s"
            from="5"
            repeatCount="indefinite"
            to="5"
            values="5;50;50;5"
          />
          <animate
            attributeName="cx"
            begin="0s"
            calcMode="linear"
            dur="2.2s"
            from="27"
            repeatCount="indefinite"
            to="27"
            values="27;49;5;27"
          />
        </circle>
        <circle
          cx="49"
          cy="50"
          r="5"
        >
          <animate
            attributeName="cy"
            begin="0s"
            calcMode="linear"
            dur="2.2s"
            repeatCount="indefinite"
            values="50;50;5;50"
          />
          <animate
            attributeName="cx"
            begin="0s"
            calcMode="linear"
            dur="2.2s"
            from="49"
            repeatCount="indefinite"
            to="49"
            values="49;5;27;49"
          />
        </circle>
      </g>
    </g>
  </svg>
</div>
`;
